fs/unstable_types.ts

```diff
@@ -201,3 +201,383 @@ export interface WriteFileOptions {
+   * based APIs.
+   *
+   * @example Usage
+   * ```ts ignore
+   * import { open } from "@std/fs/unstable-open";
+   * using file = await open("my_file.txt", { read: true });
+   * const decoder = new TextDecoder();
+   * for await (const chunk of file.readable) {
+   *   console.log(decoder.decode(chunk));
+   * }
+   * ```
+   */
+  readable: ReadableStream<Uint8Array>;
+  /**
+   * A {@linkcode WritableStream} instance to write the contents of the
+   * file. This makes it easy to interoperate with other web streams based
+   * APIs.
+   *
+   * @example Usage
+   * ```ts ignore
+   * import { open } from "@std/fs/unstable-open";
+   * const items = ["hello", "world"];
+   * using file = await open("my_file.txt", { write: true });
+   * const encoder = new TextEncoder();
+   * const writer = file.writable.getWriter();
+   * for (const item of items) {
+   *   await writer.write(encoder.encode(item));
+   * }
+   * ```
+   */
+  writable: WritableStream<Uint8Array>;
+  /**
+   * Close the file. Closing a file when you are finished with it is
+   * important to avoid leaking resources.
+   *
+   * @example Usage
+   * ```ts ignore
+   * import { open } from "@std/fs/unstable-open";
+   * using file = await open("my_file.txt");
+   * // do work with "file" object
+   * ```
+   */
+  close(): void;
+  /** Disposal protocol enabling the `using` keyword **TypeScript Only** */
+  [Symbol.dispose](): void;
+  /**
+   * **UNSTABLE**: New API, yet to be vetted.
+   *
+   * Checks if the file resource is a TTY (terminal).
+   *
+   * @example Usage
+   * ```ts ignore
+   * // This example is system and context specific
+   * import { open } from "@std/fs/unstable-open";
+   * using file = await open("/dev/tty6");
+   * file.isTerminal(); // true
+   * ```
+   */
+  isTerminal(): boolean;
+  /**
+   * Acquire an advisory file-system lock for the file.
+   *
+   * @default {false}
+   */
+  lock(exclusive?: boolean): Promise<void>;
+  /**
+   * Synchronously acquire an advisory file-system lock synchronously for the
+   * file.
+   *
+   * @default {false}
+   */
+  lockSync(exclusive?: boolean): void;
+  /**
+   * Read the file into an array buffer (`p`).
+   *
+   * Resolves to either the number of bytes read during the operation or EOF
+   * (`null`) if there was nothing more to read.
+   *
+   * It is possible for a read to successfully return with `0` bytes. This
+   * does not indicate EOF.
+   *
+   * **It is not guaranteed that the full buffer will be read in a single
+   * call.**
+   *
+   * @example Usage
+   * ```ts ignore
+   * import { open } from "@std/fs/unstable-open";
+   * // if "/foo/bar.txt" contains the text "hello world":
+   * using file = await open("/foo/bar.txt");
+   * const buf = new Uint8Array(100);
+   * const numberOfBytesRead = await file.read(buf); // 11 bytes
+   * const text = new TextDecoder().decode(buf);  // "hello world"
+   * ```
+   */
+  read(p: Uint8Array): Promise<number | null>;
```

Comment:
@ry Do you think we should port this API as well to `@std/fs`?

Reply:
Context: This is legacy API from when Deno have 2 sets of I/O interfaces Reader/Writer vs Web Streams. We deleted Reader/Writer interfaces in many places at Deno 2, but we kept this API for backward compatiblity (to my understanding).

Reply:
Ok. Let's keep them